### ISSUE TITLE
Importing GLTF doesn't specify interpolation for aiNodeAnim type

### ISSUE THREAD

**Issue by m0tholith** [assimp/assimp#5668]
**Describe the technical debt**
Animation types in the .gltf importer are not exposed (at least not to the C API).

**Better solution**
I suggest adding an `mInterpolation` field to the `aiNodeAnim` class, and using `Interpolation_LINEAR` as a default value for filetypes that don't support specifying the type of interpolation; that way different animation channels can use different interpolation methods (mimicking how .gltf specifies interpolation types).

**Desktop (please complete the following information):**
 - OS: NixOS 24.05
 - Browser: firefox
 - Version: 5.4

**Additional context**
This issue is pretty much a bump of #4890 since I'm actually trying to use the exact same file that issue's OP was trying to use.

**Comment by kimkulling** [comment]
This is true for the whole interface. SO all loaders a missing this information. There is the implicite rule to use linear interpolation, which is not good enough from my point of view.

**Comment by tellypresence** [comment]
Feature request issues tracked via epic #6183

**Comment by Chang-Jin-Lee** [comment]
This looks resolved on master, and I think it can be closed.

`aiAnimInterpolation` landed the day after your comment here, in `da281b7f`. It went onto `aiVectorKey` and `aiQuatKey` as a per-key field, one level below the `aiNodeAnim` this issue proposed. That turned out to be the more expressive home: COLLADA specifies interpolation per key, so a channel-level field would have had to discard that.

glTF started populating it in #6543 (`8ef1461c`, February), which maps the sampler mode and keeps the CUBICSPLINE tangents. Both commits are in v6.0.5.

I checked this on current master rather than reading it off the diff — three minimal files, one per sampler mode, imported and dumped:

```
STEP         -> key 0 Step,       key 1 Step
LINEAR       -> key 0 Linear,     key 1 Linear
CUBICSPLINE  -> 6 keys, all CubicSpline (3 per keyframe: in-tangent, value, out-tangent)
```

`mInterpolation` is declared `C_ENUM aiAnimInterpolation`, so it reaches the C API as well, which was the original complaint in #4890.

One gap is left behind. `grep -rn mInterpolation test/unit/` finds nothing, so none of this has regression coverage. I am happy to send a test if that would be useful.
